File: unit-test/test-sample.java
import org.junit.jupiter.api.*;

class NameOfModuleToTestTest {

  // Define top-level test variables here

    @BeforeAll
    static void beforeAll() {
    // One-time initialization logic _if required_
    }

    @BeforeEach
    void beforeEach() {
    // Logic that must be started before every test _if required_
    }

    @AfterAll
    static void afterAll() {
    // Logic that must be started after all tests _if required_
    }

    @Nested
    class MethodNameTests {
    // Define method-level variable here

        // Use method-level @BeforeAll, @BeforeEach or @AfterAll _if required_

        @Test
        void testCase() {
      // Test case code

      // to assert definitions of variables use:
            // Assertions.assertNotNull(variable);

      // to assert equality use:
            // Assertions.assertEquals(expected, actual);

            // for exceptions use:
            // Assertions.assertThrows(ExpectedException.class, () -> { /* code */ });
        }
    }
}

File: unit-test/test-sample.ts

// import necessary methods from an abstract testing framework
import { describe, expect, it, beforeAll, beforeEach, afterAll } from './setup.js';

describe('<NAME_OF_MODULE_TO_TEST>', () => {
  // Define top-level test variables here

  beforeAll(async () => {
    // One-time initialization logic _if required_
  });

  beforeEach(async () => {
    // Logic that must be started before every test _if required_
  });

  afterAll(async () => {
    // Logic that must be started after all tests _if required_
  });

  describe('#<METHOD_NAME>', () => {
    // Define method-level variable here

    // Use method-lavel beforeAll, beforeEach or afterAll _if required_

    it('<TEST_CASE>', async () => {
      // Test case code

      // to assert definitions of variables use:
      // expect(<VARIABLE>).toBeDefined();

      // to assert equality use:
      // expect(<TEST_RESULT>).toEqual(<EXPECTED_VALUE>);
      // expect(<TEST_RESULT>).toStrictEqual(<EXPECTED_VALUE>);

      // for promises use async assertion:
      // await expect(<ASYNC_METHOD>).rejects.toThrow(<ERROR_MESSAGE>);
      // await expect(<ASYNC_METHOD>).resolves.toEqual(<EXPECTED_VALUE>);
    });
  });
});